### PR TITLE
Category and single video

### PR DESCRIPTION
**Pull request by MuskanChhatrasal**
Completed
1. Single Video Page
2. Watch later page with functionality
3.  History page with functionality

### CODE REVIEW

File: src/Components/Sidebar/sidebar.js [hunk 1]
```diff
@@ -23,6 +23,15 @@ const Sidebar = () => {
...
+        <div className="list-topic side-tags" style={{ fontSize: "1.35rem" }}>
+          <Link to="/videos">
+            <i class="fas fa-video" style={{ marginRight: "0.75rem" }}></i>
+            <span>Videos</span>
+          </Link>
+        </div>
+      </div>
+
       <div>
         <div className="list-topic side-tags" style={{ fontSize: "1.35rem" }}>
```
Comment (aayushjain-code) [inline src/Components/Sidebar/sidebar.js:36]: Try minimizing use of inline syling

File: src/Components/VideoCard/videoCard.js [hunk 2]
```diff
@@ -1,34 +1,109 @@
...
+    watchLaterVideos,
+  } = useWatchLater();
+
+  const { addVideoToHistory, removeVideoFromHistory } = useHistory();
+
+  useEffect(() => {
+    getWatchLaterVideos();
+  }, []);
+
+  // const handleDropDown = () => {
```
Comment (aayushjain-code) [inline src/Components/VideoCard/videoCard.js:30]: Can remove commented Code

File: src/Components/VideoCard/videoCard.js [hunk 3]
```diff
@@ -1,34 +1,109 @@
...
+import React, { useEffect, useState } from "react";
+import { Link, useLocation } from "react-router-dom";
+import { useHistory } from "../../Context/historyContext";
+import { useWatchLater } from "../../Context/watchLaterContext";
 
 const VideoCard = ({ video }) => {
   const [dropdown, setDropdown] = useState(false);
+
+  const location = useLocation();
   const trimHeading = (word, n) => {
```
Comment (aayushjain-code) [inline src/Components/VideoCard/videoCard.js:10]: You Can create a utility export for such functions and can reduce the code repetition by importing from single place;
` export const trimHeading = (word, n) => {
    if (word.length > n) {
      return word.substring(0, n - 3) + "...";
    }
    return word;
  };`
  
  and Import it here!

File: src/Pages/Home/home.js [hunk 4]
```diff
@@ -32,23 +33,72 @@ const Home = () => {
...
+              Learn Best Food Receipes with top
+            </p>
+            <p className="font-bold text-3xl">videos from WOW TUBE</p>
+            <Link className="btn-hero font-bold text-base" to="/videos">
+              Videos
+            </Link>
+          </div>
+        </div>
+        <div className="spacer-3"></div>
+        <h2 style={{ color: "white", fontWeight: "500" }} className="spacer-4">
```
Comment (aayushjain-code) [inline src/Pages/Home/home.js:63]: You Can declare Similar styling in one place!

**Review by aayushjain-code — APPROVED** [review]
With a Few Optimizations Added, You can carry forward!

File: src/Pages/Home/home.js (hunk 4 above)
Comment (MuskanChhatrasal) [reply src/Pages/Home/home.js:63]: Okay. Will do that. Thankyou for reviewing

**Review by Developers-codz — APPROVED** [review]
Code and UI both looks good to me
Everything already suggested just work on that

**Comment by MuskanChhatrasal** [comment]
> Code and UI both looks good to me Everything already suggested just work on that

Okayy. Thankyou

File: src/Components/VideoCard/videoCard.js [hunk 5]
```diff
@@ -38,10 +60,29 @@ const VideoCard = ({ video }) => {
...
           <ul className="card-dropdown">
-            <li>
-              <i className="fas fa-clock" style={{ marginRight: "0.5rem" }}></i>
-              Add to watch later
-            </li>
+            {watchLaterVideos.some((it) => it._id === video._id) ? (
+              <li onClick={() => removeItemFromWatchLater(video._id)}>
+                <i
+                  className="fas fa-clock"
+                  style={{ marginRight: "0.5rem" }}
```
Comment (MayankKumar10) [inline src/Components/VideoCard/videoCard.js:71]: Suggestion: try to avoid inline styling

File: src/Context/watchLaterContext.js [hunk 6]
```diff
@@ -48,6 +48,7 @@ const WatchLaterProvider = ({ children }) => {
   };
 
   const addItemToWatchLater = async (video) => {
+    console.log("videos:", video);
```
Comment (MayankKumar10) [inline src/Context/watchLaterContext.js:51]: Suggestion: remove console log

File: src/Pages/WatchLater/watchLater.js [hunk 7]
```diff
@@ -1,13 +1,122 @@
...
+
+  useEffect(() => {
+    getWatchLaterVideos();
+  }, []);
   return (
-    <div>
+    <div className="watchLater-flex">
       <Sidebar />
+
+      {/* <div className="video-flex">
```
Comment (MayankKumar10) [inline src/Pages/WatchLater/watchLater.js]: Suggestion: remove the comment from here

File: src/Pages/WatchLater/watchLater.js [hunk 8]
```diff
@@ -1,13 +1,122 @@
...
+          <h1>No Videos added to Watch Later</h1>
+        )}
+      </div>
     </div>
   );
 };
 
 export default WatchLater;
+
+// <div className="main-container">
```
Comment (MayankKumar10) [inline src/Pages/WatchLater/watchLater.js]: Suggestion: Remove this unnecessary code.

**Review by MayankKumar10 — COMMENTED** [review]
Look into these Suggestions everything else looks good to me you can merge it.

**Pull request merged**